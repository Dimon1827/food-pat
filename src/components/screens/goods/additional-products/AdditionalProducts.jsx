import ProductCard from '../catalog/product-cart/ProductCard';
import './AdditionalProducts.scss';
import goodsWithDiscount from './goodsWithDiscount';

function AdditionalProducts() {
  return (
    <section class="additional-products">
      <div class="container">
        <h2 class="additional-products__title title">Сделки дня</h2>
        <div class="additional-products__wrapper">
          {goodsWithDiscount.map((good) => <ProductCard {...good}/>)}
  
          {/* <div class="additional-products__card product-card">
            <a href="404.html" class="product-card__link">
              <picture>
                <source
                  type="image/webp"
                  srcset="./day-item/day-item-1.webp"
                  width="280"
                  height="240"
                />
                <img
                  class="additional-products__image product-card__image"
                  src="images/day-item-1.png"
                  alt="Семена Органическое киноа, Коричневый и красный рис"
                  width="280"
                  height="240"
                />
              </picture>
            </a>
            <ul class="additional-products__list">
              <li class="additional-products__item">
                <span class="additional-products__number">3</span>
                <p class="additional-products__text">дня</p>
              </li>
              <li class="additional-products__item">
                <span class="additional-products__number">11</span>
                <p class="additional-products__text">часов</p>
              </li>
              <li class="additional-products__item">
                <span class="additional-products__number">18</span>
                <p class="additional-products__text">минут</p>
              </li>
            </ul>
            <div class="product-card__wrapper">
              <div class="product-card__top-wrapper">
                <a href="404.html" class="product-card__link">
                  <h2 class="product-card__subtitle subtitle">
                    Салат с фруктами
                  </h2>
                </a>
                <button class="product-card__button button">Купить</button>
              </div>
              <p class="product-card__text">Ингредиенты:</p>
              <p class="product-card__ingredients">
                Апельсин, зелень, огурцы, банан
              </p>
              <p class="product-card__text">Цена:</p>
              <span class="food__definition additional-products__old-price">
                500 ₽
              </span>
              <span class="food__definition food__price">400 ₽</span>
            </div>
          </div>
          <div class="additional-products__card product-card">
            <a href="404.html" class="product-card__link">
              <picture>
                <source
                  type="image/webp"
                  srcset="images/day-item-2.webp"
                  width="280"
                  height="240"
                />
                <img
                  class="additional-products__image product-card__image"
                  src="images/day-item-2.png"
                  alt="Сок из апельсина"
                  width="280"
                  height="240"
                />
              </picture>
            </a>
            <ul class="additional-products__list">
              <li class="additional-products__item">
                <span class="additional-products__number">5</span>
                <p class="additional-products__text">дней</p>
              </li>
              <li class="additional-products__item">
                <span class="additional-products__number">01</span>
                <p class="additional-products__text">часов</p>
              </li>
              <li class="additional-products__item">
                <span class="additional-products__number">30</span>
                <p class="additional-products__text">минут</p>
              </li>
            </ul>
            <div class="product-card__wrapper">
              <div class="product-card__top-wrapper">
                <a href="404.html" class="product-card__link">
                  <h2 class="product-card__subtitle subtitle">
                    Сок из апельсина
                  </h2>
                </a>
                <button class="product-card__button button">Купить</button>
              </div>
              <p class="product-card__text">Ингредиенты:</p>
              <p class="product-card__ingredients">Апельсин, грейфрукт</p>
              <p class="product-card__text">Цена:</p>
              <span class="food__definition additional-products__old-price">
                200 ₽
              </span>
              <span class="food__definition food__price">120 ₽</span>
            </div>
          </div>
          <div class="additional-products__card product-card">
            <a href="404.html" class="product-card__link">
              <picture>
                <source
                  type="image/webp"
                  srcset="images/day-item-3.webp"
                  width="280"
                  height="240"
                />
                <img
                  class="additional-products__image product-card__image"
                  src="images/day-item-3.png"
                  alt="Салат с лимонным коктелем"
                  width="280"
                  height="240"
                />
              </picture>
            </a>
            <ul class="additional-products__list">
              <li class="additional-products__item">
                <span class="additional-products__number">6</span>
                <p class="additional-products__text">дней</p>
              </li>
              <li class="additional-products__item">
                <span class="additional-products__number">18</span>
                <p class="additional-products__text">часов</p>
              </li>
              <li class="additional-products__item">
                <span class="additional-products__number">14</span>
                <p class="additional-products__text">минут</p>
              </li>
            </ul>
            <div class="product-card__wrapper">
              <div class="product-card__top-wrapper">
                <a href="404.html" class="product-card__link">
                  <h2 class="product-card__subtitle subtitle">
                    Салат с лимонным коктелем
                  </h2>
                </a>
                <button class="product-card__button button">Купить</button>
              </div>
              <p class="product-card__text">Ингредиенты:</p>
              <p class="product-card__ingredients">Горох, лимон, апельсин</p>
              <p class="product-card__text">Цена:</p>
              <span class="food__definition additional-products__old-price">
                450 ₽
              </span>
              <span class="food__definition food__price">350 ₽</span>
            </div>
          </div>
          <div class="additional-products__card product-card">
            <a href="404.html" class="product-card__link">
              <picture>
                <source
                  type="image/webp"
                  srcset="images/day-item-4.webp"
                  width="280"
                  height="240"
                />
                <img
                  class="additional-products__image product-card__image"
                  src="images/day-item-4.png"
                  alt="Еда для кота со вкусом курицы на 500 грамм"
                  width="280"
                  height="240"
                />
              </picture>
            </a>
            <ul class="additional-products__list">
              <li class="additional-products__item">
                <span class="additional-products__number">7</span>
                <p class="additional-products__text">дней</p>
              </li>
              <li class="additional-products__item">
                <span class="additional-products__number">04</span>
                <p class="additional-products__text">часов</p>
              </li>
              <li class="additional-products__item">
                <span class="additional-products__number">17</span>
                <p class="additional-products__text">минут</p>
              </li>
            </ul>
            <div class="product-card__wrapper">
              <div class="product-card__top-wrapper">
                <a href="404.html" class="product-card__link">
                  <h2 class="product-card__subtitle subtitle">
                    Сок из грейфрукта
                  </h2>
                </a>
                <button class="product-card__button button">Купить</button>
              </div>
              <p class="product-card__text">Ингредиенты:</p>
              <p class="product-card__ingredients">
                Грейфрукт, апельсин, банан, яблоко
              </p>
              <p class="product-card__text">Цена:</p>
              <span class="food__definition additional-products__old-price">
                220 ₽
              </span>
              <span class="food__definition food__price">140 ₽</span>
            </div>
          </div> */}
        </div>
      </div>
    </section>
  );
}

export default AdditionalProducts;
