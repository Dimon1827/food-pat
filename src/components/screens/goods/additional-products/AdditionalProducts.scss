@import '@/assets/styles/variables';

.additional-products {
  position: relative;

  .additional-products__title {
    font-size: 24px;
    line-height: 24px;
    margin-bottom: 200px;
    background-color: $white-color;
    display: inline-block;
    padding-right: 20px;

    &::before {
      content: "";
      display: block;
      position: absolute;
      top: 14px;
      left: 21px;
      width: 93%;
      height: 2px;
      background-color: $bg-basic-orange;
      z-index: -2;
    }

    @media (min-width: $tablet-width) {
      border: none;
      padding: 0 41px 0 40px;
      font-size: 36px;
      line-height: 36px;

      &::before {
        top: 21px;
        left: 0;
        width: 100%;
      }
    }

    @media (min-width: $desktop-width) {
      margin-left: -40px;
      padding-left: 40px;

      &::before {
        top: 23px;
        left: -80px;
        width: 106%;
      }
    }
  }
}



.additional-products__wrapper {
  margin: 0 auto;
  margin-bottom: 50px;

  @media (min-width: $tablet-width) {
    display: grid;
    justify-content: center;
    grid-template-columns: repeat(2, 45%);
    gap: 180px 15px;
  }

  @media (min-width: $desktop-width) {
    grid-template-columns: repeat(4, 250px);
    gap: 180px 63px;
  }
}

.additional-products__list {
  margin: 0 auto;
  list-style-type: none;
  position: absolute;
  top: -69px;
  left: 0;
  right: 0;
  padding: 0;
  display: flex;
  width: 85%;
  justify-content: center;
}

.additional-products__number {
  display: block;
  font-size: 20px;
  text-align: center;
  color: #3bb77e;
  margin-bottom: 5px;
}

.additional-products__item {
  background-color: $white-color;
  padding: 9px;
  border-radius: 4px;

  &:not(:last-child) {
    margin-right: 30px;
  }
}

.additional-products__text {
  margin: 0;
  font-size: 16px;
  line-height: 100%;
  text-transform: capitalize;
  color: #7e7e7e;
}

.additional-products__card {
  position: relative;
  width: 85%;

  &:not(:last-child) {
    margin-bottom: 200px;
  }

  @media (min-width: $tablet-width) {
    &:not(:last-child) {
      margin-bottom: 0;
    }
  }

  @media (min-width: $desktop-width) {
    width: 100%;
  }
}

.additional-products__old-price {
  border: none;
  text-decoration: line-through;
}

.additional-products__image {
  position: absolute;
  top: -165px;
  left: -25px;
  z-index: -1;
  width: 120%;
  max-width: 465px;
}

