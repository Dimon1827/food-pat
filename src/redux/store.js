import { configureStore, combineReducers } from '@reduxjs/toolkit';
import {
  persistReducer,
  persistStore,
  FLUSH,
  REHYDRATE,
  PAUSE,
  PERSIST,
  PURGE,
  REGISTER,
} from 'redux-persist';
import storage from 'redux-persist/lib/storage';
import productsReducer from './slice/productsSlice';

const persistConfig = {
  key: 'root',
  version: 2,
  storage,
};

const reducer = combineReducers({
  products: productsReducer,
});

const persistedReducer = persistReducer(persistConfig, reducer);

const store = configureStore({
  reducer: persistedReducer,
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware({
      serializableCheck: {
        ignoredActions: [FLUSH, REHYDRATE, PAUSE, PERSIST, PURGE, REGISTER],
      },
    }),
});

export const persistor = persistStore(store);

export default store;
