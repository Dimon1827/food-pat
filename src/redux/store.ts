import { configureStore, combineReducers } from '@reduxjs/toolkit';
import { useDispatch } from 'react-redux';
import {
  persistReducer,
  persistStore,
  FLUSH,
  REHYDRATE,
  PAUSE,
  PERSIST,
  PURGE,
  REGISTER,
} from 'redux-persist';
import storage from 'redux-persist/lib/storage';
import productsReducer from './slice/productsSlice';

const persistConfig = {
  key: 'root',
  version: 2,
  storage,
};

const reducer = combineReducers({
  products: productsReducer,
});

const persistedReducer = persistReducer(persistConfig, reducer);

const store = configureStore({
  reducer: persistedReducer,
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware({
      serializableCheck: {
        ignoredActions: [FLUSH, REHYDRATE, PAUSE, PERSIST, PURGE, REGISTER],
      },
    }),
});

export type RootState = ReturnType<typeof persistedReducer>;
export type AppDispatch = typeof store.dispatch;
export const useAppDispatch = () => useDispatch<AppDispatch>();
export const persistor = persistStore(store);
export default store;
