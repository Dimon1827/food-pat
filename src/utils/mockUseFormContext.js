import { useFormContext } from "react-hook-form";

export function mockUseFormContext(errors) {
  useFormContext.mockReturnValue({
    register: jest.fn(),
    formState: {
      errors: errors,
    },
  });
}


