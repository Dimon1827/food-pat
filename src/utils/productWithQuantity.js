const productWithQuantity = (food) => {
  return {
    ...food,
    isChecked: false,
    quantity: 1,
  };
};

export default productWithQuantity;
