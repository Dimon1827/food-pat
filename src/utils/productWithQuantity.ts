import { IProducts } from "redux/slice/productsSlice.types";

interface IproductWithQuantity extends Omit<IProducts, 'isChecked' | 'quantity'> {};

const productWithQuantity = (food: IproductWithQuantity) => {
  return {
    ...food,
    isChecked: false,
    quantity: 1,
  };
};

export default productWithQuantity;
