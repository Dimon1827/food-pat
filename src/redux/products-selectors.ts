import { RootState } from "./store";

export const selectProducts = (state: RootState) => state.products.products;
export const selectTotalPrice = (state: RootState) => state.products.totalPrice;
export const selectTotalQuantity = (state: RootState) => state.products.totalQuantity;
