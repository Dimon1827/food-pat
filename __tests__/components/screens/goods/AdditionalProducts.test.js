import AdditionalProducts from '@/components/screens/goods/additional-products/AdditionalProducts';
import { renderWithRouter } from '@/utils/renderWithRouter';
import * as store from '@/redux/store';

const mockedDispatch = jest.spyOn(store, 'useAppDispatch');

describe('Additional products', () => {
  it('ProductCard renders correctly', () => {
    mockedDispatch.mockReturnValue(jest.fn());
    const { getByTestId } = renderWithRouter(<AdditionalProducts />);
    expect(getByTestId('additional-products')).toMatchSnapshot();
  });
});
