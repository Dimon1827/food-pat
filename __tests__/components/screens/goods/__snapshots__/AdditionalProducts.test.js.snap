// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Additional products ProductCard renders correctly 1`] = `
<section
  class="additional-products"
  data-testid="additional-products"
>
  <div
    class="container"
  >
    <h2
      class="additional-products__title title"
    >
      Сделки дня
    </h2>
    <div
      class="additional-products__wrapper"
    >
      <div
        class="product-card additional-products__card"
        data-testid="product-card"
      >
        <a
          href="/Салат с фруктами"
        >
          <picture>
            <source
              height="240"
              srcset="./assets/images/day-item-1.webp"
              type="image/webp"
              width="280"
            />
            <img
              alt="Салат с фруктами"
              class="product-card__image additional-products__image"
              height="240"
              src="./assets/images/day-item-1.png"
              width="280"
            />
          </picture>
        </a>
        <ul
          class="additional-products__list"
        >
          <li
            class="additional-products__item"
          >
            <span
              class="additional-products__number"
            >
              3
            </span>
            <p
              class="additional-products__text"
            >
              дня
            </p>
          </li>
          <li
            class="additional-products__item"
          >
            <span
              class="additional-products__number"
            >
              11
            </span>
            <p
              class="additional-products__text"
            >
              часов
            </p>
          </li>
          <li
            class="additional-products__item"
          >
            <span
              class="additional-products__number"
            >
              18
            </span>
            <p
              class="additional-products__text"
            >
              минут
            </p>
          </li>
        </ul>
        <div
          class="product-card__wrapper"
          data-testid="product-card-information"
        >
          <div
            class="product-card__top-wrapper"
          >
            <a
              class="product-card__link"
              href="/Салат с фруктами"
            >
              <h2
                class="product-card__subtitle subtitle"
              >
                Салат с фруктами
              </h2>
            </a>
            <button
              class="product-card__button button"
            >
              Купить
            </button>
          </div>
          <p
            class="product-card__text"
          >
            Ингредиенты:
          </p>
          <p
            class="product-card__ingredients"
          >
            Апельсин, зелень, огурцы, банан
          </p>
          <p
            class="product-card__text"
          >
            Цена:
          </p>
          <span
            class="food__definition additional-products__old-price"
          >
            500
             ₽
          </span>
          <span
            class="food__definition food__price"
          >
            400
             ₽
          </span>
        </div>
      </div>
      <div
        class="product-card additional-products__card"
        data-testid="product-card"
      >
        <a
          href="/Сок из апельсина"
        >
          <picture>
            <source
              height="240"
              srcset="./assets/images/day-item-2.webp"
              type="image/webp"
              width="280"
            />
            <img
              alt="Сок из апельсина"
              class="product-card__image additional-products__image"
              height="240"
              src="./assets/images/day-item-2.png"
              width="280"
            />
          </picture>
        </a>
        <ul
          class="additional-products__list"
        >
          <li
            class="additional-products__item"
          >
            <span
              class="additional-products__number"
            >
              5
            </span>
            <p
              class="additional-products__text"
            >
              дня
            </p>
          </li>
          <li
            class="additional-products__item"
          >
            <span
              class="additional-products__number"
            >
              12
            </span>
            <p
              class="additional-products__text"
            >
              часов
            </p>
          </li>
          <li
            class="additional-products__item"
          >
            <span
              class="additional-products__number"
            >
              30
            </span>
            <p
              class="additional-products__text"
            >
              минут
            </p>
          </li>
        </ul>
        <div
          class="product-card__wrapper"
          data-testid="product-card-information"
        >
          <div
            class="product-card__top-wrapper"
          >
            <a
              class="product-card__link"
              href="/Сок из апельсина"
            >
              <h2
                class="product-card__subtitle subtitle"
              >
                Сок из апельсина
              </h2>
            </a>
            <button
              class="product-card__button button"
            >
              Купить
            </button>
          </div>
          <p
            class="product-card__text"
          >
            Ингредиенты:
          </p>
          <p
            class="product-card__ingredients"
          >
            Апельсин, грейфрукт
          </p>
          <p
            class="product-card__text"
          >
            Цена:
          </p>
          <span
            class="food__definition additional-products__old-price"
          >
            200
             ₽
          </span>
          <span
            class="food__definition food__price"
          >
            120
             ₽
          </span>
        </div>
      </div>
      <div
        class="product-card additional-products__card"
        data-testid="product-card"
      >
        <a
          href="/Салат с лимонным коктелем"
        >
          <picture>
            <source
              height="240"
              srcset="./assets/images/day-item-3.webp"
              type="image/webp"
              width="280"
            />
            <img
              alt="Салат с лимонным коктелем"
              class="product-card__image additional-products__image"
              height="240"
              src="./assets/images/day-item-3.png"
              width="280"
            />
          </picture>
        </a>
        <ul
          class="additional-products__list"
        >
          <li
            class="additional-products__item"
          >
            <span
              class="additional-products__number"
            >
              6
            </span>
            <p
              class="additional-products__text"
            >
              дня
            </p>
          </li>
          <li
            class="additional-products__item"
          >
            <span
              class="additional-products__number"
            >
              18
            </span>
            <p
              class="additional-products__text"
            >
              часов
            </p>
          </li>
          <li
            class="additional-products__item"
          >
            <span
              class="additional-products__number"
            >
              14
            </span>
            <p
              class="additional-products__text"
            >
              минут
            </p>
          </li>
        </ul>
        <div
          class="product-card__wrapper"
          data-testid="product-card-information"
        >
          <div
            class="product-card__top-wrapper"
          >
            <a
              class="product-card__link"
              href="/Салат с лимонным коктелем"
            >
              <h2
                class="product-card__subtitle subtitle"
              >
                Салат с лимонным коктелем
              </h2>
            </a>
            <button
              class="product-card__button button"
            >
              Купить
            </button>
          </div>
          <p
            class="product-card__text"
          >
            Ингредиенты:
          </p>
          <p
            class="product-card__ingredients"
          >
            Горох, лимон, апельсин
          </p>
          <p
            class="product-card__text"
          >
            Цена:
          </p>
          <span
            class="food__definition additional-products__old-price"
          >
            450
             ₽
          </span>
          <span
            class="food__definition food__price"
          >
            350
             ₽
          </span>
        </div>
      </div>
      <div
        class="product-card additional-products__card"
        data-testid="product-card"
      >
        <a
          href="/Сок из грейфрукта"
        >
          <picture>
            <source
              height="240"
              srcset="./assets/images/day-item-4.webp"
              type="image/webp"
              width="280"
            />
            <img
              alt="Сок из грейфрукта"
              class="product-card__image additional-products__image"
              height="240"
              src="./assets/images/day-item-4.png"
              width="280"
            />
          </picture>
        </a>
        <ul
          class="additional-products__list"
        >
          <li
            class="additional-products__item"
          >
            <span
              class="additional-products__number"
            >
              7
            </span>
            <p
              class="additional-products__text"
            >
              дня
            </p>
          </li>
          <li
            class="additional-products__item"
          >
            <span
              class="additional-products__number"
            >
              14
            </span>
            <p
              class="additional-products__text"
            >
              часов
            </p>
          </li>
          <li
            class="additional-products__item"
          >
            <span
              class="additional-products__number"
            >
              17
            </span>
            <p
              class="additional-products__text"
            >
              минут
            </p>
          </li>
        </ul>
        <div
          class="product-card__wrapper"
          data-testid="product-card-information"
        >
          <div
            class="product-card__top-wrapper"
          >
            <a
              class="product-card__link"
              href="/Сок из грейфрукта"
            >
              <h2
                class="product-card__subtitle subtitle"
              >
                Сок из грейфрукта
              </h2>
            </a>
            <button
              class="product-card__button button"
            >
              Купить
            </button>
          </div>
          <p
            class="product-card__text"
          >
            Ингредиенты:
          </p>
          <p
            class="product-card__ingredients"
          >
            Грейфрукт, апельсин, банан, яблоко
          </p>
          <p
            class="product-card__text"
          >
            Цена:
          </p>
          <span
            class="food__definition additional-products__old-price"
          >
            220
             ₽
          </span>
          <span
            class="food__definition food__price"
          >
            140
             ₽
          </span>
        </div>
      </div>
    </div>
  </div>
</section>
`;
